pkg/ibmvpc/vpc.go

```diff
@@ -361,66 +350,85 @@ func (n *NACL) GeneralConnectivityPerSubnet(subnet *Subnet) string {
 
 func subnetFromNode(node vpcmodel.Node) (subnet *Subnet, err error) {
 	switch concreteNode := node.(type) {
-	case *NetworkInterface:
-		return concreteNode.subnet, nil
-	case *IKSNode:
-		return concreteNode.subnet, nil
-	case *ReservedIP:
-		return concreteNode.subnet, nil
+	case vpcmodel.InternalNodeIntf:
+		return concreteNode.Subnet().(*Subnet), nil
 	default:
 		return nil, fmt.Errorf("cannot get subnet for node: %+v", node)
 	}
 }
 
+type naclConnectivityInput struct {
+	targetNode           vpcmodel.Node
+	nodeInSubnet         vpcmodel.Node
+	subnet               *Subnet
+	subnetAffectedByNACL bool
+	targetWithinSubnet   bool
+}
+
```

Comment:
This is yet another optimization not related to the specific abstraction added in this PR. right?

Reply:
right

Reply:
next time it would be nice to have it in a different PR :-)